src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -178,6 +188,60 @@ public DRSObject lookupObjectByDrsId(AuthenticatedUserRequest authUser, String d
         }
     }
 
+    public DRSAccessURL getAccessUrlForObjectId(AuthenticatedUserRequest authUser, String objectId, String accessId) {
+        DRSObject object = lookupObjectByDrsId(authUser, objectId, false);
+
+        DrsId drsId = drsIdService.fromObjectId(objectId);
+        Snapshot snapshot = snapshotService.retrieve(UUID.fromString(drsId.getSnapshotId()));
+        SnapshotProject snapshotProject = snapshotService.retrieveAvailableSnapshotProject(snapshot.getId());
+        GoogleProjectResource projectResource = snapshot.getProjectResource();
+
+        FSItem fsObject = null;
+        try {
+            String lookupTimer = performanceLogger.timerStart();
+            fsObject = fileService.lookupSnapshotFSItem(
+                snapshotProject,
+                drsId.getFsObjectId(),
+                1);
+
+            performanceLogger.timerEndAndLog(
+                lookupTimer,
+                objectId, // not a flight, so no job id
+                this.getClass().getName(),
+                "fileService.lookupSnapshotFSItem");
+        } catch (InterruptedException ex) {
+            throw new FileSystemExecutionException("Unexpected interruption during file system processing", ex);
+        }
+
+        GcsPdao.GcsLocator locator = GcsPdao.getGcsLocatorFromGsPath(fsObject.getPath());
+        String gsBucket = locator.getBucket();
+
+        if (doesAccessIdMatch(accessId, object)) {
```

Comment:
I don't understand why this code needs to both get the `DRSObject` for the ID and call `fileService.lookupSnapshotFSItem()`. I would expect that the `DRSObject` contains the information this method needs already.